orion/controllers/programs/programsController.go

```diff
@@ -81,6 +79,40 @@ func DeleteProgram(c *gin.Context) {
   return
 }
 
-func CheckValidProgram() bool {
+func CheckValidProgram(c *gin.Context) bool {
+  // Retrieves the inputted values
+  name := c.Param("programName")
+  grade1 := c.Param("grade1")
+  grade2 := c.Param("grade2")
+
+  // Checks if program name is empty
+  if name == "" {
+    c.String(http.StatusBadRequest, "Invalid name " + name)
+    return false
+  }
+
+  // Checks if the program name is a string only
+  for _, i := range name {
+    if (i < 'a' || i > 'z') && (i < 'A' || i > 'Z') && i != '_' {
```

Comment:
good check. But this program name could also have numbers as well. I wonder if we can use a regex (regular expression) to do this check? Have you ever done one of these before?

Reply:
I’ve done regex before. If it can include numbers, then is this check necessary? I thought the variable was already a string.

Reply:
I think it's still worth it. We want to check all alphanumeric character values and underscores. no special characters like -.?*^ etc.